### DISCUSSION TITLE
GDS support for array types or struct types as the types of parameters

### DISCUSSION THREAD

**Discussion by shalevm** [nasa/fprime#1596] [Q&A] [answered]
Hi,

the user guide currently states that 

> The F Prime GDS does not currently support the use of array types or struct types as the types of parameters.

Is this still the case? 
if so - is support for this expected  any time soon? if not - is there a recommended work-around (at least for array types)?

thanks,
Shalev.

**Comment by ThibFrgsGmz** [comment, marked as answer]
I haven't tried again in a while but I haven't seen any PR about this GDS limitation. The ticket for this problem is still open in `fprime-gds`: https://github.com/fprime-community/fprime-gds/issues/38

**Reply by shalevm** [reply]
thanks.
for now i just don't use structs (flatten them if needed) and break array to separate fields. since i can script this pretty easily, it's no biggie.

**Reply by LeStarch** [reply]
This is correct.  We have yet to support parameters/commands with struct and array types.  Flattening for commands is the recommended work around.
